src/routing/http-routing-v1.md

```diff
@@ -106,6 +106,60 @@ Each object in the `Providers` list is a record conforming to a schema, usually
 
 ## Peer Routing API
 
+### `GET /routing/v1/closest-peers/{peer-id}?[closerThan]&[count]`
+
+#### Path Parameters
+
+- `peer-id` is a [Peer ID](https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md) represented as a CIDv1 encoded with `libp2p-key` codec.
+
+#### Query Paramters
+
+- `closerThan` is an optional [Peer ID](https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md) represented as a CIDv1 encoded with `libp2p-key` codec.
+  - Returned peer records must be closer to `peer-id` than `closerThan`.
+  - If omitted the routing implementation should use it's own [Peer ID](https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md).
+- `count` is an optional number that specifies how many peer records the requester desires.
+  - Minimum 1, maximum 100, default 20.
```

Comment:
If the `closerThan` parameter is set, and `count` is omitted, the default value should be the `maximum`, otherwise some peers closer than `closerThan` may be filtered out. Both parameters could however be combined if both are set.

Reply:
I think my assumption here is that the peers would be sorted by closeness value before the limit was applied.

Would this not be the case?

Reply:
> I think my assumption here is that the peers would be sorted by closeness value before the limit was applied.

Yes that is correct. My point is that both `closerThan` and `count` can limit the count of peers that are returned, the parameters may conflict with each other.

E.g `peer-id=key0&closerThan=key1` should return all the peers between `key0` and `key1`, but if there are more than `default=20` peers between `key0` and `key1` this query will only return the `default` closest, and not all of them up to the `maximum`.

If none of `count` or `closerThan` parameters is set, the default `count` should be `default=20`. However, if `closerThan` is set and not `count`, the default `count` should be set to the `maximum=100`. This allows to return as many peers as possible between `key0` and `key1`, and if the caller wants to set a limit, they should combine `closerThan` and `count`.

Also, when setting `count` and not `closerThan` using the node's peer id as default `closerThan` parameter can be problematic. E.g I want to provide a CID to the 20 closest nodes, and I am the 12th closest peer to this CID. I will receive only 11 closer nodes, and will be unable to find the `20` closest nodes. Hence, if `count` is set and not `closerThan` the `closerThan` default should be the opposite of the target `peer-id` (`peer-id XOR 111...11`). Maybe the opposite of the target `peer-id` is a better default for `closerThan` than the requester's peer id, allowing the default response to contain a constant number of peers.